core/src/storage/collections/binary_heap/access_wrapper.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Provides a wrapper around `SyncChunk` which stores a defined amount
+//! of values in one cell (instead of the `SyncChunk` behavior of storing
+//! one value per cell). The intention is to reduce expensive fetch
+//! operations from storage.
+
+use crate::storage::{
+    chunk::SyncChunk,
+    Flush,
+};
+use scale::{
+    Decode,
+    Encode,
+};
+
+// Number of values stored in each entry of the `SyncChunk`.
+const COUNT: u32 = 3;
+
+#[derive(Copy, Clone, Debug, Encode, Decode)]
+pub struct Group<T> ([Option<T>; COUNT as usize]);
+
+#[derive(Debug, Encode, Decode)]
+pub struct AccessWrapper<T> (SyncChunk<Group<T>>);
```

Comment:
Could we maybe rename this to something less generic? Maybe `DuplexSyncChunk`.
Also in ink! we generally prefer speaking field names with normal structs towards tuple-structs usage.
However, I think this use case is okay since it is really just a wrapper.